src/oidcop/construct.py

```diff
@@ -35,43 +41,45 @@ def assign_algorithms(typ):
-            try:
-                _proposal = kwargs[attr]
-            except KeyError:
-                if default_val is not None:
-                    _info[attr] = default_val
-                elif "signing_alg_values_supported" in attr:
-                    _info[attr] = assign_algorithms("signing_alg")
-                    if attr == "token_endpoint_auth_signing_alg_values_supported":
-                        # none must not be in
-                        # token_endpoint_auth_signing_alg_values_supported
-                        if "none" in _info[attr]:
-                            _info[attr].remove("none")
-                elif "encryption_alg_values_supported" in attr:
-                    _info[attr] = assign_algorithms("encryption_alg")
-                elif "encryption_enc_values_supported" in attr:
-                    _info[attr] = assign_algorithms("encryption_enc")
-            else:
-                _permitted = None
-
-                if "signing_alg_values_supported" in attr:
-                    _permitted = set(assign_algorithms("signing_alg"))
-                elif "encryption_alg_values_supported" in attr:
-                    _permitted = set(assign_algorithms("encryption_alg"))
-                elif "encryption_enc_values_supported" in attr:
-                    _permitted = set(assign_algorithms("encryption_enc"))
-
-                if _permitted and not _permitted.issuperset(set(_proposal)):
-                    raise ValueError(
-                        "Proposed set of values outside set of permitted ({})".__format__(
-                            attr
-                        )
-                    )
+    if default_capabilities is None:
+        return default_capabilities
+
+    _info = {}
+    for attr, default_val in default_capabilities.items():
+        if attr in kwargs:
+            _proposal = kwargs[attr]
+            _permitted = None
 
-                _info[attr] = _proposal
-        return _info
-    else:
-        return None
+            if "signing_alg_values_supported" in attr:
+                _permitted = set(assign_algorithms("signing_alg"))
+            elif "encryption_alg_values_supported" in attr:
+                _permitted = set(assign_algorithms("encryption_alg"))
+            elif "encryption_enc_values_supported" in attr:
+                _permitted = set(assign_algorithms("encryption_enc"))
+
+            if _permitted and not _permitted.issuperset(set(_proposal)):
+                raise ValueError(
+                    "Proposed set of values outside set of permitted, "
+                    f"'{attr}' sould be {_permitted} it's instead {_proposal}"
+                )
+            _info[attr] = _proposal
+        else:
+            if default_val is not None:
+                _info[attr] = default_val
+            elif "signing_alg_values_supported" in attr:
+                _info[attr] = assign_algorithms("signing_alg")
+                if "none" in _info[attr]:
```

Comment:
@roland this is the only warning I'd tell you.
In you previour approach only if attr == "token_endpoint_auth_signing_alg_values_supported" the none value has been removed. I didn't understad this condition, so I remove none at every occourrence.

Feel free to revert and explain why that rule is only valid for `token_endpoint_auth_signing_alg_values_supported`